opennms-alarms/daemon/src/test/java/org/opennms/netmgt/alarmd/AlarmdIT.java

```diff
@@ -548,9 +545,86 @@ public void testArchiveAlarm() throws Exception {
         assertThat(m_alarmDao.findAll().stream().filter(a -> !a.isArchived()
                 && !OnmsSeverity.CLEARED.equals(a.getSeverity())).count(), equalTo(1L));
     }
+    
+    @Test
+    public void testDualAlarmState() throws Exception {
+        AlarmPersisterImpl persisterImpl = (AlarmPersisterImpl)m_alarmd.getPersister();
+        
+        // Enable the legacy two alarm state functionality
+        persisterImpl.setLegacyAlarmState(true);
+
+        final MockNode node = m_mockNetwork.getNode(1);
+
+        // There should be no alarms in the alarms table
+        assertEmptyAlarmTable();
+
+        //this should be the first occurrence of this alarm
+        //there should be 1 alarm now
+        String dbname = m_database.getTestDatabase();
+        System.out.println(dbname);
+        sendNodeDownEvent(node);
+
+        // Wait until we've create the node down alarm
+        Callable<Integer> numAlarmsCallable = getNumAlarmsCallable();
+        await().atMost(10, SECONDS).until(numAlarmsCallable, equalTo(1));
+
+        // Send in the UP
+        sendNodeUpEvent(node);
+
+        // We should have two alarms now
+        numAlarmsCallable = getNumAlarmsCallable();
+        await().atMost(10, SECONDS).until(numAlarmsCallable, equalTo(2));
+
+        Thread.sleep(1000);
```

Comment:
If we're waiting for the alarm to be cleared, we should use `await()....` with that condition instead of an explicit sleep.

Reply:
I tried... will try that again.

Reply:
√ thanks for the help.